src/Optimizely/OptimizelyConfig/OptimizelyConfigService.php

```diff
@@ -80,15 +96,103 @@ public function getConfig()
     {
         $experimentsMaps = $this->getExperimentsMaps();
         $featuresMap = $this->getFeaturesMap($experimentsMaps[1]);
-
+        $attributes = $this->getConfigAttributes();
+        $audiences = $this->getConfigAudiences();
+        $events = $this->getConfigEvents();
         return new OptimizelyConfig(
             $this->revision,
             $experimentsMaps[0],
             $featuresMap,
-            $this->datafile
+            $this->datafile,
+            $this->environmentKey,
+            $this->sdkKey,
+            $attributes,
+            $audiences,
+            $events
         );
     }
     
+    /**
+     * Generates array of attributes as OptimizelyAttribute.
+     *
+     *
+     * @return array of OptimizelyAttributes.
+     */
+    protected function getConfigAttributes()
+    {
+        $attributArray = [];
+        $attributes = $this->projectConfig->getAttributes();
+        foreach($attributes as $attr){
+            $optly_attr = new OptimizelyAttribute(
+                $attr['id'],
+                $attr['key']
+            );
+            array_push($attributArray, $optly_attr);
+        }
+        return $attributArray;
+    }
+
+    /**
+     * Generates array of events as OptimizelyEvents.
+     *
+     *
+     * @return array of OptimizelyEvents.
+     */
+    protected function getConfigEvents()
+    {
+        $eventsArray = [];
+        $events = $this->projectConfig->getEvents();
+        foreach($events as $event){
+            $optly_event = new OptimizelyEvent(
+                $event['id'],
+                $event['key'],
+                $event['experimentIds']
+            );
+            array_push($eventsArray, $optly_event);
+        }
+        return $eventsArray;
+    }
+
+    /**
+     * Generates array of audiences giving typed audiences high priority as OptimizelyAudience.
+     *
+     *
+     * @return array of OptimizelyEvents.
+     */
+    protected function getConfigAudiences()
+    {
+        $optAudiences = [];
+        $uniqueIds = [];
```

Comment:
use a map here instead of array to keep track of unique ids, that way you wont have to iterate them for searching.